src/schema/schema_validators.py

```diff
@@ -626,6 +625,73 @@ def validate_group_name(property_key, normalized_entity_type, request, existing_
+
+    if  'superseded_associated_processed_component_uuids' in existing_data_dict \
+        and 'superseded_associated_processed_component_uuids' in new_data_dict:
+        raise ValueError(   f"'superseded_associated_processed_component_uuids' is already set on"
+                            f" {existing_data_dict['uuid']}.")
+    if  'new_associated_multi_assay_uuid' in existing_data_dict \
+        and 'new_associated_multi_assay_uuid' in new_data_dict:
+        raise ValueError(   f"'new_associated_multi_assay_uuid' is already set on"
+                            f" {existing_data_dict['uuid']}.")
+    if  'superseded_associated_processed_component_uuids' in new_data_dict \
+        and 'new_associated_multi_assay_uuid' in new_data_dict:
+        raise ValueError(   f"'superseded_associated_processed_component_uuids' and 'new_associated_multi_assay_uuid'"
+                            f" cannot both be specified on a single Dataset.")
+    if  'superseded_associated_processed_component_uuids' in new_data_dict \
+        and 'new_associated_multi_assay_uuid' in existing_data_dict:
+        raise ValueError( f"'superseded_associated_processed_component_uuids' cannot be set on"
+                        f" existing Dataset {existing_data_dict['uuid']} because it is a component Dataset of"
+                        f" {existing_data_dict['new_associated_multi_assay_uuid']}.")
+    if  'new_associated_multi_assay_uuid' in new_data_dict \
+        and 'superseded_associated_processed_component_uuids' in existing_data_dict:
+        # Convert the string from Neo4j the Python list
+        supersededComponentDatasets = schema_manager.convert_str_literal(existing_data_dict['superseded_associated_processed_component_uuids'])
+        raise ValueError( f"'new_associated_multi_assay_uuid' cannot be set on"
+                        f" existing Dataset {existing_data_dict['uuid']} because it is a Multi-Assay Dataset"
+                        f" with {len(supersededComponentDatasets)}"
+                        f" component Datasets it supersedes.")
+    # If no contradictions above have caused a ValueError, check if new data contains UUIDs for valid entities.
+    if  'new_associated_multi_assay_uuid' in new_data_dict:
+        proposedMultiAssayDataset = schema_neo4j_queries.get_entity(schema_manager.get_neo4j_driver_instance()
+                                                                    , new_data_dict['new_associated_multi_assay_uuid'])
+        if len(proposedMultiAssayDataset) < 1:
+            raise ValueError(   f"'new_associated_multi_assay_uuid' value"
+                                f" {new_data_dict['new_associated_multi_assay_uuid']}"
+                                f" does not exist.")
+    if  'superseded_associated_processed_component_uuids' in new_data_dict:
+        proposedComponentsList = schema_manager.convert_str_literal(new_data_dict['superseded_associated_processed_component_uuids'])
```

Comment:
We don't need to use `convert_str_literal()` here since the uuids list is already a Python list from converted from request JSON payload.